ComputerVision/measurements.py

```diff
@@ -0,0 +1,170 @@
+            change_image()
+        else :
+            messagebox.showerror( "Error", "There is already a line here.")
+    else :
+       messagebox.showerror( "Error", "You need 2 points to create a calculation.")
+
+def add_measurement_buttons () :
+    settings.variables['cellid'] = 1
+    settings.frames['buttons'] = tk.Frame(master=settings.m, background="white")
+    settings.frames['buttons'].grid(row=0, column=3, sticky="n", rowspan=5)
+    settings.buttons['undoButton'] = tk.Button(master=settings.frames['buttons'], text="Undo Action", command=undo, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['undoButton'].pack()
+    settings.buttons['calculateButton'] = tk.Button(master=settings.frames['buttons'], text="Calculate", command=calculate, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['calculateButton'].pack()
+    settings.buttons['hideIDsButton'] = tk.Button(master=settings.frames['buttons'], text="Hide IDs", command=hide_ids, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['hideIDsButton'].pack()
+    settings.buttons['printButton'] = tk.Button(master=settings.frames['buttons'], text="Preview Measurements", command=preview, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['printButton'].pack()
+    settings.buttons['downloadButton'] = tk.Button(master=settings.frames['buttons'], text="Download Measurements", command=download, width=30, height=2, padx=5, pady=5, background="white")
+    settings.buttons['downloadButton'].pack()
+
+def draw_circle (x, y) :
+    settings.previousImgs.append([settings.images["imagetomeasure"].copy(),"circle"])
+    settings.images["imagetomeasure"] = cv2.circle(settings.images["imagetomeasure"],(x-2,y-2),5,(255,0,0),-1)
+    img = cv2.circle(settings.previousImgsWithoutIDs[len(settings.previousImgsWithoutIDs)-1][0],(x-2,y-2),5,(255,0,0),-1)
+    settings.previousImgsWithoutIDs.append([img])
+    change_image()
+
+    settings.points.append([x-2, y-2, True])
+
+
+def getorigin(eventorigin):
+    global x, y
+    x = eventorigin.x
+    y = eventorigin.y
+    draw_circle(x, y)
+
+def measures_with_canny () :
+    screen_width = int(settings.m.winfo_screenwidth()/1.25)
+    screen_height = int(settings.m.winfo_screenheight()/1.25)
+    settings.images["imagetomeasure"] = cv2.resize(cv2.cvtColor(settings.images["latestCanny"],cv2.COLOR_GRAY2RGB), (screen_width, screen_height), interpolation = cv2.INTER_AREA)
+    settings.previousImgsWithoutIDs.append([settings.images["imagetomeasure"].copy()])
+    settings.frames["originalimage"].destroy()
+    settings.imagelabels["measurementimage"] = settings.imagelabels["altered"]
+    settings.imagelabels["measurementimage"].bind("<Button-1>",getorigin)
+
+    settings.frames["sliderframe"].destroy()
+    settings.frames["endCannyframe"].destroy()
+    change_image()
+    add_measurement_buttons()
+
+def measures_with_original () :
```

Comment:
the measures_with_canny differes from measures_with_original  by one word which is the key in the settings.images dictionary, make it one function

Reply:
done!